homeassistant/components/ps4/__init__.py

```diff
@@ -32,3 +34,40 @@
     await hass.config_entries.async_forward_entry_unload(
         entry, 'media_player')
     return True
+
+
+async def async_migrate_entry(hass, entry):
+    """Migrate old entry."""
+    from pyps4_homeassistant.media_art import COUNTRIES
+
+    config_entries = hass.config_entries
+    data = entry.data
+    version = entry.version
+
+    reason = {1: "Region codes have changed"}  # From 0.89
+
+    # Migrate Version 1 -> Version 2
+    if version == 1:
+        loc = await hass.async_add_executor_job(location.detect_location_info)
+        if loc:
+            country = loc.country_name
+            if country in COUNTRIES:
+                for device in data['devices']:
+                    device[CONF_REGION] = country
+                entry.version = 2
+                config_entries.async_update_entry(entry, data=data)
+                _LOGGER.info(
+                    "PlayStation 4 Config Updated: \
+                    Region changed to: %s", country)
+                return True
+
+    msg = """{} for the PlayStation 4 Integration.
+            Please remove the PS4 Integration and re-configure
```

Comment:
Will the entry and devices still work in this case until it's removed and reconfigured? Should we remove it automatically?

Reply:
Yes it should still work and not break. Media art will be slower to retrieve and may be inaccurate however so it would be advised to remove but not entirely necessary.

Reply:
I'm not sure. What's the best user experience?

Reply:
Well If we remove the integration automatically the user may not remember to re-add or realize what had happened unless we can make a notification persist across restart. Since the version migration is non-breaking I think displaying the notification only is the best way to address this. Entity will still work and user will be reminded that they should remove and  re-configure until it is addressed.